src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2652,6 +2671,9 @@ protected virtual void MarkInstruction (Instruction instruction, MethodDefinitio
 				}
 
 			case OperandType.InlineType:
+				if (instruction.OpCode.Code == Code.Newarr) {
```

Comment:
Are you sure this is the only relevant opcode? What about code like `var foos = (IFoo[]) Array.CreateInstance (typeof (Foo), 1);` ?

Reply:
That pattern should be fine because the `typeof(Foo)` means "visible to reflection" and that's handled elsewhere in this pull request.